### PR TITLE
Issue 6835 - Adding ids option support for hasMany/belongsToMany associations

### PR DESCRIPTION
**Pull request by gaetansnl**
#6835

Example

``` php
        $entity = $this->Users->patchEntity(
            $entity, 
            $this->request->data,[
            'associated'=>[
            'Tags'=>[
                    'ids' => true
                ]
            ]
        ]);
```

Only tags in _ids param will be linked. Any other array will be ignored.

``` php
        $data = [
        ........
            'tags' =>[
                '_ids' =>[....]
            ]
        ]
```

### CODE REVIEW

**Comment by markstory** [comment]
You have some [phpcs errors](https://travis-ci.org/cakephp/cakephp/jobs/67509281).

File: src/ORM/Marshaller.php [hunk 1]
```diff
@@ -223,12 +223,20 @@ protected function _marshalAssociation($assoc, $value, $options)
         if (in_array($assoc->type(), $types)) {
             return $marshaller->one($value, (array)$options);
         }
+        if ($assoc->type() === Association::ONE_TO_MANY || $assoc->type() === Association::MANY_TO_MANY) {
+            $hasIds = array_key_exists('_ids', $value);
+            $idsOption = array_key_exists('ids', $options) && $options['ids'];
```
Comment (markstory) [inline src/ORM/Marshaller.php]: This new option needs to be documented.

**Comment by pulse14** [comment]
I've also added documentation for `_ids` format, for marshalling and merging.

**Comment by markstory** [comment]
Obsoleted by #7347

**Pull request closed without merging**